### PR TITLE
cargo-unfmt: init at 0.3.3

### PR DESCRIPTION
**Pull request by cafkafk**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ca/cargo-unfmt/package.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  buildInputs = lib.optionals stdenv.isDarwin [ libiconv darwin.apple_sdk.frameworks.Security ];
+
+  doCheck = false;
+
+  outputs = [ "out" ];
+
+  meta = with lib; {
+    description = "Unformat code into perfect rectangles";
+    license = licenses.gpl3;
```
Comment (Aleksanaa) [inline pkgs/by-name/ca/cargo-unfmt/package.nix:30]:
```suggestion
    license = licenses.gpl3Plus;
```

File: pkgs/by-name/ca/cargo-unfmt/package.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  cargoHash = "sha256-mMeHTYCUIZR3jVvTxfyH4I9wGfUdCWcyn9djnksAY8k=";
+
+  buildInputs = lib.optionals stdenv.isDarwin [ libiconv darwin.apple_sdk.frameworks.Security ];
+
+  doCheck = false;
+
+  outputs = [ "out" ];
+
+  meta = with lib; {
+    description = "Unformat code into perfect rectangles";
```
Comment (Aleksanaa) [inline pkgs/by-name/ca/cargo-unfmt/package.nix:29]:
```suggestion
    description = "Unformat code into perfect rectangles";
    homepage = "https://github.com/fprasx/cargo-unfmt";
```

File: pkgs/by-name/ca/cargo-unfmt/package.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    license = licenses.gpl3;
+    mainProgram = "cargo-unfmt";
+    maintainers = with maintainers; [ cafkafk ];
+    platforms = platforms.unix ++ platforms.windows;
```
Comment (Aleksanaa) [inline pkgs/by-name/ca/cargo-unfmt/package.nix:33]:
```suggestion
```

I'd recommend removing this line so every platform that has rust support can use it. Mentioning windows here looks weird.

File: pkgs/by-name/ca/cargo-unfmt/package.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    hash = "sha256-nvn4nZkkNQQvzShwoxtFqHeyhXQPm2GJoTKBI+MkFgM=";
+  };
+
+  cargoHash = "sha256-mMeHTYCUIZR3jVvTxfyH4I9wGfUdCWcyn9djnksAY8k=";
+
+  buildInputs = lib.optionals stdenv.isDarwin [ libiconv darwin.apple_sdk.frameworks.Security ];
+
+  doCheck = false;
+
+  outputs = [ "out" ];
```
Comment (Aleksanaa) [inline pkgs/by-name/ca/cargo-unfmt/package.nix:26]:
```suggestion
```

There is no need to specify in this case

File: pkgs/by-name/ca/cargo-unfmt/package.nix [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    repo = "cargo-unfmt";
+    rev = "0f4882f65d248e32812e0e854fa11d7db60921e7";
+    hash = "sha256-nvn4nZkkNQQvzShwoxtFqHeyhXQPm2GJoTKBI+MkFgM=";
+  };
+
+  cargoHash = "sha256-mMeHTYCUIZR3jVvTxfyH4I9wGfUdCWcyn9djnksAY8k=";
+
+  buildInputs = lib.optionals stdenv.isDarwin [ libiconv darwin.apple_sdk.frameworks.Security ];
+
+  doCheck = false;
```
Comment (Aleksanaa) [inline pkgs/by-name/ca/cargo-unfmt/package.nix:24]: Why is this disabled? Please provide a reason in the comment near it.

**Comment by cafkafk** [comment]
@Aleksanaa ty for the quick review :3

Addressed your comments with latest force push.

File: pkgs/by-name/ca/cargo-unfmt/package.nix [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+  buildInputs = lib.optionals stdenv.isDarwin [ libiconv darwin.apple_sdk.frameworks.Security ];
+
+  # Doc tests are broken on 0.3.3
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Unformat code into perfect rectangles";
+    homepage = "https://github.com/fprasx/cargo-unfmt";
+    license = licenses.gpl3plus;
```
Comment (Aleksanaa) [inline pkgs/by-name/ca/cargo-unfmt/package.nix:30]:
```suggestion
    license = licenses.gpl3Plus;
```

Comment (cafkafk) [reply pkgs/by-name/ca/cargo-unfmt/package.nix]: Ohh missed that, thanks, pushed a fix now!

**Pull request merged**